src/main/webapp/app/overview/exercise-chatbot/exercise-chatbot.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component } from '@angular/core';
+import { faCommentDots } from '@fortawesome/free-solid-svg-icons';
+import { MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { ChatbotPopupComponent } from './chatbot-popup/chatbot-popup.component';
+import { ExerciseChatWidgetComponent } from 'app/overview/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component';
+
+@Component({
+    selector: 'jhi-exercise-chatbot',
+    templateUrl: './exercise-chatbot.component.html',
+    styleUrls: ['./exercise-chatbot.component.scss'],
+})
+export class ExerciseChatbotComponent {
+    public chatAccepted = 'false';
+    public buttonDisabled = false;
+    private dialogRef: MatDialogRef<ExerciseChatWidgetComponent> | null = null;
+
+    // Icons
+    faCommentDots = faCommentDots;
+
+    constructor(private dialog: MatDialog) {}
+
+    ngOnDestroy() {
```

Comment:
You've got a valid client style warning here. Please fix this.
> Lifecycle interface 'OnDestroy' should be implemented for method 'ngOnDestroy'. (https://angular.io/styleguide#style-09-01)

Reply:
fixed